src/shared/install.c

```diff
@@ -66,6 +69,31 @@ typedef struct {
         OrderedHashmap *have_processed;
 } InstallContext;
 
+typedef struct {
+        const char *pattern;
```

Comment:
sounds strange to make this const I must say. I think const mostly makes sense for stuff that is located in actual read-only memory, or in function signatures. But I doubt this makes sense here, in particular as you actually have to cast he const away in order to free the string.